File: code_samples/haskell/utilities_date_time.hs
-- Utilities: Date Time
-- AI/ML Training Sample

module DateTime where

data DateTime = DateTime {
    getData :: String
} deriving (Show, Eq)

process :: DateTime -> String -> DateTime
process obj input = obj { getData = input }

validate :: DateTime -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = DateTime { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)

File: code_samples/ruby/utilities_date_time.rb
# Utilities: Date Time
# AI/ML Training Sample

class DateTime
  attr_accessor :data
  
  def initialize
    @data = ""
  end
  
  def process(input)
    @data = input
  end
  
  def validate
    !@data.empty?
  end
  
  def to_s
    "Data: #{@data}"
  end
end

# Example usage
instance = DateTime.new
instance.process("example")
puts instance
puts "Valid: #{instance.validate}"
